### ISSUE TITLE
Minified.js redefines global define&require functions if only require was defined

### ISSUE THREAD

**Issue by bedrin** [timjansen/minified.js#60]
Some libraries provide only a `require` function, without `define`
Minified.js breaks such libraries. Consider following example:
http://jsfiddle.net/bedrin/g6kyho7v/

As you can see, I can use Shred before adding Minified.JS, but after that it is broken.

The problematic code is this:
```javascript
if (/^u/.test(typeof define)) { // no AMD support available ? define a minimal version
	(function(def){
		var require = this['require'] = function(name) { return def[name]; };
		this['define'] = function(name, f) { def[name] = def[name] || f(require); };
	})({});
}
```

It should test for `require` function as well

**Comment by timjansen** [comment]
Thank you, will add that additional check.

Tim

**Comment by timjansen** [comment]
Looked at it again, and I don't see a good way of solving this issue in Minified. If I add the check for 'require', Minified will go into AMD-compatibility mode. But that won't work either, because Shred's 'require' definition would clash with an AMD loader. 

Minified is designed to either work with an existing AMD loader, or provide its own stub. I am not sure what Shred is using... adding CommonJS support would be an option for Minified, but from what I have seen in the (minified) Shred source code, it won't help either. As long as Shred just declares global variables without supporting any module system, it's hard to avoid naming clashes.

**Comment by bedrin** [comment]
Yeah, I wanted to make a pull request but also didn't find a good way to solve it.

As a workaround I have added following javascript before embedding (basically copy-pasteing) Minified to my code:
```javascript
;(function(){
    var defs = {};

    function require(name) {
        return defs[name];
    }

    function define(name, f) {
        defs[name] = defs[name] || f(require);
    }

    //@@include('../bower_components/minified/dist/minified.js')

    // my code which is using minified
}.apply({}));
```

With this code in front of Minified, it doesn't spoil the global (window) context.

Probably you can think of changing the Minified so it would define 'require' and 'define' functions not in global scope but rather in local scope, so it would be manageable by end users
